enumerable.rb

```diff
@@ -0,0 +1,86 @@
+# rubocop:disable Style/For
+
+# Enumerable module
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    return unless block_given?
+
+    array = []
+    my_each { |x| array << x if yield x }
+
+    array
+  end
+
+  def my_all?
+    return unless block_given?
+
+    result = my_select { |x| yield x }
+    length == result.length
+  end
+
+  def my_any?
+    return unless block_given?
+
+    result = my_select { |x| yield x }
+    result.length.positive?
+  end
+
+  def my_none?
+    return unless block_given?
+
+    result = my_select { |x| yield x }
+    !result.length.positive?
+  end
+
+  def my_count(count = nil)
+    return count if count
+    return length unless block_given?
+
+    my_select { |x| yield x }.length
+  end
+
+  def my_map(my_proc = nil)
+    array = []
+    my_each { |x| array << my_proc.call(x) } if my_proc
+    my_each { |x| array << yield(x) } if block_given?
+
+    array
+  end
```

Comment:
- [ ] returns an Enumerator if no block is given

- [ ] executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.